remarkable_update_image/image.py

```diff
@@ -347,6 +364,17 @@ def signature(self):
         # TODO - get from entry
         return None
 
+    @property
+    def version(self) -> str | None:
+        return self._version
+
+    @property
+    def hardware_type(self) -> str:
+        return self._hardware_type
+
+    def __getitem__(self, key):
+        return self._archive[key]
```

Comment:
I'm not a fan of this interface, and I'd rather it be more explicitly just exposing the archive.
```suggestion
    @property
    def archive(self):
        return self._archive
```